receiver.py

```diff
@@ -37,13 +37,22 @@ def data_lines(self):
 class Receiver:
 
     def __init__(self, db_type):
-        self.data_shape = struct.Struct(
-            ''.join(map(lambda x: x[0], db_type)))
-        self.data_size = self.data_shape.size
+        self.data_shape = [struct.Struct(
```

Comment:
This is no longer a `data_shape` it is a list of possible data shapes. Name it accordingly.